HW-3 mvvm (Delegates.observable) 변환

ddaeyeonkim opened the pull request.

ddaeyeonkim committed:
HW 3 아키텍처 변경을 위해 TickerContract, TickerPresenter 제거

ddaeyeonkim committed:
HW 3 TickerFragment 내부에 TickerViewModel 클래스 생성 및 구현

namjackson commented on improve777/GAASProject/app/src/main/java/dev/daeyeon/gaasproject/ui/ticker/TickerFragment.kt:
```diff
@@ -165,34 +165,97 @@ class TickerFragment : Fragment(), TickerContract.View {
+                ResponseCode.CODE_NULL_FAIL_MSG -> activity!!.toast(R.string.all_network_error)
+                ResponseCode.CODE_EMPTY_SUCCESS -> activity!!.toast(R.string.ticker_fragment_empty)
+                else -> activity!!.toast(newValue)
+            }
```
뷰모델이 이너클래스로 정의되어 있어서 애매한데, 
뷰모델에선 newValue만 던지고 
실제 메세지 처리는 View쪽에서 처리해주는 게 좋을거같아요!

ddaeyeonkim replied:
수정 완료하였습니다

2b5062a5803f06a37ea6d30d33ae56bbd89b561f

namjackson commented on improve777/GAASProject/app/src/main/java/dev/daeyeon/gaasproject/ui/ticker/TickerFragment.kt:
```diff
@@ -165,34 +165,97 @@ class TickerFragment : Fragment(), TickerContract.View {
+                return@observable
+            }
+            binding.srlTicker.isRefreshing = newValue
+        }
```
이부분도 마찬가지로 뷰쪽에 잇는게 더 좋을거같아요!

ddaeyeonkim replied:
수정 완료하였습니다

2b5062a5803f06a37ea6d30d33ae56bbd89b561f

sjjeong replied:
토스트는 빼는게 맞는데 이거는 여기서 처리해도 됩니다.

ddaeyeonkim replied:
수정 완료하였습니다

5936872dea02d706f677df3efce1bc6696143608

namjackson commented on improve777/GAASProject/app/src/main/java/dev/daeyeon/gaasproject/ui/ticker/TickerFragment.kt:
```diff
@@ -165,34 +165,97 @@ class TickerFragment : Fragment(), TickerContract.View {
+         * 기준 마켓
+         */
+        private lateinit var _baseCurrency: String
+
```
변수 네이밍에 대해서 찾아봤는데 , 아래와 같이 변수명이 중복되는 경우에 _변수로 선언하고 
그외에는 변수앞에 _를  빼는게 좋다고합니다! 


Names for backing properties
If a class has two properties which are conceptually the same but one is part of a public API and another is an implementation detail, use an underscore as the prefix for the name of the private property:

`
class C {
      private val _elementList = mutableListOf<Element>()
      
      val elementList: List<Element>
            get() = _elementList
}`

[https://kotlinlang.org/docs/reference/coding-conventions.html#naming-rules](url)

ddaeyeonkim replied:
수정 완료하였습니다

4595d6095fa8cbdd0cb936d0eb412fd0fe1075cf

ddaeyeonkim committed:
HW 3 kotlin convention 의 권장사양에 맞추어 private 필드 언더바 제거

ddaeyeonkim committed:
HW 3 view 동작 코드를 viewModel 에서 activity 의 메소드로 분리 후 호출

namjackson reviewed (dismissed):
수고하셨습니당!

ddaeyeonkim committed:
HW 3 showProgress() / hideProgress() 제거 후 뷰모델에서 프로그레스 상태 변경 호출

ddaeyeonkim committed:
HW 3 BaseFragment 생성 및 범위지정함수(run) 을 감싸는 bind() 메서드 추가 및 구현

TickerFragmentBinding 을 받는 bind() 메소드와 별개로
TickerFragment 내부에 선언된 searchDialogBinding 을 받기 위해
<T> bind() 메소드를 생성

ddaeyeonkim committed:
HW 3 기본 통화 설정 다이얼로그 에서 리스너의 파라미터 이름을 용도에 맞게 position 으로 수정

ddaeyeonkim committed:
HW 3 기본으로 동작이 되기 때문에 AlertDialog 의 버튼 이벤트에서 dismiss 호출 제거

ddaeyeonkim committed:
HW 3 loadUpbitTicker 에서 searchTicker 가 null 일 때 ALL_CURRENCY 로 수정

ddaeyeonkim committed:
HW 3 initCurrentArray() 메서드를 loadUpbitTicker() 에서 제거 및 가져올 때 호출

sjjeong commented on improve777/GAASProject/app/src/main/java/dev/daeyeon/gaasproject/ui/ticker/TickerFragment.kt:
```diff
@@ -195,4 +170,88 @@ class TickerFragment : Fragment(), TickerContract.View {
+
+            val list = arrayListOf(UpbitDataSource.ALL_CURRENCY)
+
+            if (marketList[0].isNotEmpty()) {
```
이 부분 코드 해석하는게 너무 오래걸립니다.
initCurrentArray()를 조금 더 readability하게 수정해주세요

ddaeyeonkim replied:
수정 완료하였습니다

0fdf167e98b7ed30cb3dd497a84a9bf21b82f23f

ddaeyeonkim committed:
HW 3 initCurrentArray() 메소드명 수정 및 코드 개선

sjjeong reviewed (approved):
LGTM ✊ ✌️ 👋
